src/stp/new_plays/AttackingPass.cpp

```diff
@@ -18,19 +19,26 @@
 
 namespace rtt::ai::stp::play {
 
+void AttackingPass::onInitialize() noexcept {
+    passerShot = false;
+    passingPosition = calculatePassLocation(Grid(0.15*field.getFieldWidth(), -2.5, 2 , 2.5, 5, 5)).first;
+}
+
 AttackingPass::AttackingPass() : Play() {
     startPlayInvariants.clear();
     startPlayInvariants.emplace_back(std::make_unique<invariant::NormalPlayGameStateInvariant>());
     startPlayInvariants.emplace_back(std::make_unique<invariant::BallCloseToUsInvariant>());
     startPlayInvariants.emplace_back(std::make_unique<invariant::NoGoalVisionFromBallInvariant>());
+    startPlayInvariants.emplace_back(std::make_unique<invariant::BallClosestToUsInvariant>());
```

Comment:
How cool would it be to have a polymorphic container that could just contain all this shit... too bad std::vector is limited to non-polymorphic types, then again you'd have to segment the hell out of your memory for it, or somehow keep track of sizes runtime, which is less than ideal.

Reply:
Yeah, I hope we can find some better way for this invariant structure